src/reyna/Connection/ConnectionInfo.cs

```diff
@@ -62,6 +62,76 @@ public bool Mobile
             }
         }
 
+        public bool Wifi
+        {
+            get
+            {
+                try
+                {
+                    var interfaces = NetworkInterface.GetAllNetworkInterfaces();
+                    foreach (var ni in interfaces)
+                    {
+                        if (WifiNetwork(ni))
+                        {
+                            return true;
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                }
+
+                return false;
+            }
+        }
+
+        public bool Roaming
+        {
+            get
+            {
+                try
+                {
+                    var interfaces = NetworkInterface.GetAllNetworkInterfaces();
+                    bool roamingConnected = false;
+                    bool wifiConnected = false;
+                    foreach (var ni in interfaces)
+                    {
+                        if (RoamingNetwork(ni))
+                        {
+                            if (NetworkConnected(ni))
+                            {
+                                roamingConnected = true;
+                            }
+                        }
+                        else if (WifiNetwork(ni))
+                        {
+                            if (NetworkConnected(ni))
+                            {
+                                wifiConnected = true;
+                            }
+                        }
+                    }
+
+                    return roamingConnected && !wifiConnected;
+                }
+                catch (Exception)
+                {
+                }
+
+                return false;
+            }
+        }
+
+        private static bool RoamingNetwork(INetworkInterface ni)
+        {
+            return ni.Name.ToLower(System.Globalization.CultureInfo.CurrentCulture).StartsWith("roaming", StringComparison.CurrentCulture);
+        }
+
+        private static bool WifiNetwork(INetworkInterface ni)
```

Comment:
I don't think this is a working code, there is nothing that called "wifi" in the name to check the connectivity.
In Elemez code there was away to detect network type. it was based on excluding GPRS, ActiveSync and LAN.
@Youhana-Hana